catroid/src/main/java/org/catrobat/catroid/content/Look.java

```diff
@@ -377,6 +368,25 @@ public float getDirectionInUserInterfaceDimensionUnit() {
 
 	public void setRotationMode(int mode) {
 		rotationMode = mode;
+		rotateLookDatasIfNeeded(mode);
+	}
+
+	private void rotateLookDatasIfNeeded(int mode) {
+		if (mode != Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && sprite.look.isFlipped()) {
+			flipAllLookDatas();
+			sprite.look.setFlipped(false);
+		}
+		boolean orientedLeft = sprite.look.getDirectionInUserInterfaceDimensionUnit() < 0;
+		if (mode == Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && orientedLeft) {
+			flipAllLookDatas();
+			sprite.look.setFlipped(true);
+		}
+	}
+
+	public void flipAllLookDatas() {
+		for (LookData lookData : sprite.getLookDataList()) {
```

Comment:
I think this is a nice idea since simplification is always a good thing to do. Consequently, one could transfer `flipAllLookDatas()` and `rotateLookDatasIfNeeded(int mode)` into the `Sprite` classwhere it logically also belongs in my opinion. So `protected Sprite sprite;` would be obsolete then.